src/sidecar/sidecarManager.ts

```diff
@@ -379,17 +379,57 @@ export class SidecarManager {
         // Set up the environment for the sidecar process.
         const sidecar_env = constructSidecarEnv(process.env);
 
+        const stderrPath = `${SIDECAR_LOGFILE_PATH}.stderr`;
         try {
+          // try to create a file to track any stderr output from the sidecar process
+          fs.writeFileSync(stderrPath, "");
+          const stderrFd = fs.openSync(stderrPath, "w");
           const sidecarProcess = spawn(executablePath, [], {
             detached: true,
-            stdio: "ignore",
+            // ignore stdin/stdout, stderr to the file
+            stdio: ["ignore", "ignore", stderrFd],
             env: sidecar_env,
           });
+          // close the file descriptor for stderr; child process will inherit it
+          // and write to it
+          fs.closeSync(stderrFd);
```

Comment:
If spawn() throws, will still leak the file descriptor. Need the closeSync() within an interior try / finally block?